### PR TITLE
Disable tracing when PHP is executed during RINIT

### LINKED ISSUE #1428: [BUG] [0.68.0] incompatibility with openswoole extension
Segfault 

```
$ gdb php

(gdb) run  -d extension=ddtrace --ri ddtrace

Program received signal SIGSEGV, Segmentation fault.

(gdb) bt
#0  0x00007fffeb920e8d in ddtrace_observer_fcall_init () from /usr/lib64/php/modules/ddtrace.so
#1  0x0000555555868969 in zend_observer_fcall_install (execute_data=0x7ffff7412020) at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/Zend/zend_observer.c:117
#2  _zend_observe_fcall_begin (execute_data=0x7ffff7412020) at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/Zend/zend_observer.c:162
#3  0x0000555555868c65 in zend_observer_fcall_begin (execute_data=execute_data@entry=0x7ffff7412020) at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/Zend/zend_observer.c:193
#4  0x00005555558449d0 in zend_execute (op_array=0x7ffff7487000, return_value=0x7fffffffc210) at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/Zend/zend_vm_execute.h:58536
#5  0x00005555557cb160 in zend_eval_stringl (str=<optimized out>, str_len=<optimized out>, retval_ptr=0x0, string_name=0x555556adbe40 "@swoole-src/library/constants.php")
    at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/Zend/zend_execute_API.c:1193
#6  0x00007fffef480fa4 in zend::eval (
    code="\n/**\n * This file is part of Open Swoole.\n *\n * @link     https://www.swoole.co.uk\n * @contact  hello@swoole.co.uk\n * @license  https://github.com/openswoole/library/blob/master/LICENSE\n */\n\ndeclare(s"..., filename="@swoole-src/library/constants.php") at /usr/src/debug/php-pecl-openswoole-4.9.0-1.fc35.remi.8.0.x86_64/NTS/ext-src/php_swoole_cxx.cc:39
#7  0x00007fffef48103d in php_swoole_load_library () at /usr/src/debug/php-pecl-openswoole-4.9.0-1.fc35.remi.8.0.x86_64/NTS/ext-src/php_swoole_library.h:7685
#8  0x00007fffef48678d in zm_activate_openswoole (type=<optimized out>, module_number=<optimized out>) at /usr/src/debug/php-pecl-openswoole-4.9.0-1.fc35.remi.8.0.x86_64/NTS/ext-src/php_swoole.cc:1047
#9  zm_activate_openswoole (type=<optimized out>, module_number=<optimized out>) at /usr/src/debug/php-pecl-openswoole-4.9.0-1.fc35.remi.8.0.x86_64/NTS/ext-src/php_swoole.cc:1031
#10 0x00005555557dec40 in zend_activate_modules () at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/Zend/zend_API.c:2673
#11 0x0000555555773f03 in php_request_startup () at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/main/main.c:1762
#12 0x0000555555869faf in do_cli (argc=5, argv=0x555555e10c90) at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/sapi/cli/php_cli.c:913
#13 0x000055555563ab6b in main (argc=5, argv=0x555555e10c90) at /usr/src/debug/php-8.0.14-1.fc35.remi.x86_64/sapi/cli/php_cli.c:1336
quit) 

```

### PR DESCRIPTION
**Pull request by krakjoe**
### Description

Fixes #1428 

dd-trace-php has a requirement that the runtime is properly initialized before tracing is attempted - *all* loaded extensions must have executed their RINIT stage hooks before the tracer can instrument anything safely.

This change enforces that requirement.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [x] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: ext/php8/php8/engine_hooks.c [hunk 1]
```diff
@@ -738,6 +738,11 @@ static void dd_observer_end_handler(zend_execute_data *execute_data, zval *retva
 
 zend_observer_fcall_handlers ddtrace_observer_fcall_init(zend_execute_data *execute_data) {
     zend_function *fbc = EX(func);
+
+    if (!PG(modules_activated)) {
```
Comment (SammyK) [inline ext/php8/php8/engine_hooks.c:742]: To call out a behavior difference on PHP 8: any extension that runs code in RINIT would essentially disable the tracer's ability to trace any function call that is called in RINIT due to installing NULL observer handlers. This is certainly an edge case but could cause some unexpected behavior that might be a bit tricky to debug.

I wonder if emitting a log message for these cases would help with debugging? It might have to be a one-time log message so that it is not so noisy.

File: ext/php5/php5/engine_hooks.c [hunk 2]
```diff
@@ -311,6 +311,10 @@ static int dd_exit_handler(zend_execute_data *execute_data TSRMLS_DC) {
 }
 
 static ddtrace_dispatch_t *dd_lookup_dispatch_from_fbc(zend_function *fbc TSRMLS_DC) {
+    if (!PG(modules_activated)) {
```
Comment (SammyK) [inline ext/php5/php5/engine_hooks.c:314]: Since this is a hot path, do you think it is worth wrapping this in an `UNEXPECTED`?

**Review by SammyK — COMMENTED** [review]
I think we should definitely have tests for this (run some arbitrary PHP code during RINIT) which would be possible with TEA (ZAI) SAPI. But I'm assuming tests might be blocked on #1411, right?

**Review by SammyK — APPROVED** [review]
Approving this so that it is not blocked for a hotfix release. We can follow up on any of the CR suggestions in very-near future PRs and add tests once #1411 is wrapped up and merged. 👍

File: ext/php5/php5/engine_hooks.c (hunk 2 above)
Comment (krakjoe) [reply ext/php5/php5/engine_hooks.c:314]: Not in isolation, for there to be significance, we would have to break down into more branches with the other conditions, and determine which ones are expected/not expected ...

tl;dr doesn't seem worth it.
